### PR TITLE
fix: wrong configuration/legacy for regression test workflow

### PR DESCRIPTION
**Pull request by dextermallo**
## Description

There are a misconfiguration and a legacy code for the `regression test workflow`.

## Fixes

- Update configuration.
- Remove unused folder.

## Test

I have tested it with my fork repo [here](https://github.com/dextermallo/coreruleset/actions).

### CODE REVIEW

**Comment by azurit** [comment]
@dextermallo Hi and thank you for this PR.

@fzipi or @theseion can you look at this? I'm not an expert on this part of CRS.

**Comment by theseion** [comment]
@dextermallo I don't see what you are trying to fix. Where did you encounter errors?

**Comment by dextermallo** [comment]
> @dextermallo I don't see what you are trying to fix. Where did you encounter errors?

Hi @theseion, it's a minor fix. If the `modsec_version` is set to `modsec3-nginx`, the pipeline is not available anymore

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -32,14 +32,14 @@ jobs:
 
       - name: "Run tests for ${{ matrix.modsec_version }}"
         run: |
-          mkdir -p tests/logs/${{ matrix.modsec_version }}/{nginx,apache2}
+          mkdir -p tests/logs/${{ matrix.modsec_version }}
```
Comment (theseion) [inline .github/workflows/test.yml:35]: These directories can be necessary while the web server launches, before the configuration is loaded. They are the default log directories. Web server startup may fail without them.

**Review by theseion — CHANGES_REQUESTED** [review]

**Comment by theseion** [comment]
That makes sense. It's not a case we have at the moment, because we purposefully exclude nginx from the test run (too many issues with libmodsecurity3 and our tests).

File: .github/workflows/test.yml (hunk 1 above)
Comment (dextermallo) [reply .github/workflows/test.yml]: Added back in [2e262ec](https://github.com/coreruleset/coreruleset/pull/3251/commits/2e262ecd4279a6efe35ea757e8e9440c2b49ce4c)

**Review by theseion — APPROVED** [review]

**Comment by theseion** [comment]
Thank you @dextermallo.

**Pull request merged**